src/midi/midi_mt32.cpp

```diff
@@ -723,7 +689,20 @@ void MidiHandler_mt32::Render()
 		}
 		// Grab the next buffer from backstock and populate it ...
 		playable_buffer = backstock.Dequeue();
-		soft_limiter.Process(render_buffer, FRAMES_PER_BUFFER, playable_buffer);
+
+		// Scale and copy to playable buffer
+		auto in_pos  = render_buffer.begin();
+		auto out_pos = playable_buffer.begin();
+
+		while (in_pos != render_buffer.end()) {
+			AudioFrame frame = {*in_pos++, *in_pos++};
+
+			frame.left *= INT16_MAX;
+			frame.right *= INT16_MAX;
+
+			*out_pos++ = frame.left;
+			*out_pos++ = frame.right;
+		}
```

Comment:
Oh, great catch with the multiplier.. been a while since I've been in the mt32 code and forgot it's values are between. 1 and -1. 

Now that both buffers are float (and the same size), we can swap their content (which uses move semantics) and then multiply up the result. 

Something like:

```c++
std::swap(render_buffer, playable_buffer);
for (auto & s : playable_buffer)
    s *= INT16_MAX;
```

Reply:
Yeah, good point, and that works great. Just pushed up a similar change for FluidSynth as well.